esp-hal/src/timer/systimer.rs

```diff
@@ -317,50 +311,49 @@ pub trait Unit {
         }
     }
 
-    /// Update the value returned by [Self::poll_count] to be the current value
-    /// of the counter.
-    ///
-    /// This can be used to read the current value of the timer.
-    fn update(&self) {
-        let systimer = unsafe { &*SYSTIMER::ptr() };
-        systimer
-            .unit_op(self.channel() as _)
-            .modify(|_, w| w.update().set_bit());
-    }
+    /// Reads the current counter value.
+    fn read_count(&self) -> u64 {
```

Comment:
Any objections moving this synchronization logic into `Alarm` and leaving Unit as the low level API (i.e. update and poll)?

Reply:
Not much, besides the desire to not expose footgun parts.

Reply:
Now that `read_count()` is expensive, do I have a stronger argument? 😄

Reply:
How about now, now that it's cheap again? 🙃

Reply:
Yeah I don't think I have a strong enough case anymore haha